src/bdf_utils.jl

```diff
@@ -23,3 +23,158 @@ function R!(k, ρ, cache)
     end
   end
 end
+
+# This functions takes help of D2 array to create backward differences array D
+# Ith row of D2 keeps Ith order backward differences (∇ⁱyₙ)
+function backward_diff!(cache, D, D2, k, flag=true)
+  if flag
+    if typeof(cache) <: OrdinaryDiffEqMutableCache
+      D[1] .= D2[1,1]
+    else
+      D[1] = D2[1,1]
+    end
+  end
+  for i = 2:k
+    for j = 1:(k-i+1)
+      if typeof(cache) <: OrdinaryDiffEqMutableCache
+        @. D2[i,j] = D2[i-1,j] - D2[i-1,j+1]
+      else
+        D2[i,j] = D2[i-1,j] - D2[i-1,j+1]
+      end
+    end
+    if flag
+      if typeof(cache) <: OrdinaryDiffEqMutableCache
+        D[i] .= D2[i,1]
+      else
+        D[i] = D2[i,1]
+      end
+    end
+  end
+end
+
+function mul!(cache, D, R, k)
+  @unpack tmp = cache
+  if typeof(cache) <: OrdinaryDiffEqMutableCache
+    fill!(tmp,zero(eltype(D[1])))
+  else
+    tmp = zero(eltype(D))
+  end
+  for i = 1:1
```

Comment:
`1:1`? Also, you can put these together: `for i = 1:1, j = 1:k`.

Reply:
size of `D` is `1 * 5`

Reply:
That `1 x 5` is a weird MATLABism. Make it a `Vector` or an `Adjoint{Vector}`. Example: `zeros(5)'`: notice its type.